azure-templates/make-release-steps.yml

```diff
@@ -0,0 +1,81 @@
+# Make Release Steps
+# Mark a release on the project repository, with version bump and tag,
+# and publish a release on Github
+
+parameters:
+- name: 'semver'
+  type: string
+  values:
+  - major
+  - minor
+  - patch
+- name: 'gitUsername'
+  type: string
+- name: 'gitEmail'
+  type: string
+- name: 'gitHubConnection'
+  type: string
+
+- name: 'cache_version_id'
+  type: string
+  default: $(CACHE_VERSION_ID)
+  
+steps:
+- checkout: self
+  displayName: 'Checkout'
+  clean: true
+  persistCredentials: true
+
+- script: |
+    git checkout master
+  displayName: 'Checkout master'
+      
+- task: Cache@2
+  inputs:
+    key: 'yarn-${{ parameters.cache_version_id }} | "$(Agent.OS)" | yarn.lock'
+    restoreKeys: |
+      yarn-${{ parameters.cache_version_id }} | "$(Agent.OS)"
+    path: $(YARN_CACHE_FOLDER)
+  displayName: 'Cache yarn packages'
+
+- task: UseNode@1
+  inputs:
+    version: $(NODE_VERSION)
+  displayName: 'Set up Node.js'
+    
+- script: |
+    yarn install --frozen-lockfile --no-progress --non-interactive --network-concurrency 1
+  displayName: 'Install node modules'
+
+- script: |
+    git config --global user.email "${{ parameters.gitEmail }}" && git config --global user.name "${{ parameters.gitUsername }}"
+  displayName: 'Git setup'  
+
+- script: |
+    npm version ${{ parameters.semver }} -m "Bump version to %s [skip ci]"
+  displayName: 'Version bump and tag'
+
+- script: |
+    git push --follow-tags origin master 
+  displayName: 'Push to master'
+
+- script: |
+    NEXT_VERSION=$(node -p "require('./package.json').version")
+    HEAD_SHA=$(git rev-parse HEAD)
+    TITLE="Release $NEXT_VERSION"
+    TAG="v$NEXT_VERSION"
```

Comment:
can we add some suffix to the release tags (in order to detect them among all the other possible tags)
ie. `-RELEASE` ?

Reply:
Here is not where we create the tag, this is just how we instruct the Github release task on what tag point to. At this step, tag is already created and pushed. 

Tag is actually created automatically by `npm version`. If we need more control over the format, we have a couple of options:
1. use [`git-tag-version`](https://docs.npmjs.com/cli/v6/commands/npm-version#git-tag-version) flag on the command, which blocks commit and tagging (which we have to do manually then)
1. use [tag-version-prefix](https://docs.npmjs.com/cli/v6/using-npm/config#tag-version-prefix) config, but it actually works on prefix only.

I'd go for the first, but **do we really need to do that?** I think we no longer need differentiations between released and unreleased versions, we can just find out by reading the Github release page.

Reply:
imho it's useful to have some pattern to detect release tags (ie. to trigger other events filtering on those tags)